File: frontend/src/schemas/installMode.ts
import { z } from "zod";

export const InstallModeResponseSchema = z.object({
  mode: z.enum(["docker_compose_auto", "native"]),
  docker_socket_accessible: z.boolean(),
  pg_container_data_host_path: z.string().nullable(),
});
export type InstallModeResponse = z.infer<typeof InstallModeResponseSchema>;

File: backend/app/models/api/install_mode.py
"""DTOs install_mode (LOT pairing-exec)."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class InstallModeResponse(BaseModel):
    mode: Literal["docker_compose_auto", "native"]
    docker_socket_accessible: bool
    pg_container_data_host_path: str | None
